adwords_client/client.py

```diff
@@ -276,24 +275,124 @@ def _batch_operations(self, file_name):
+
+        service = self.service(previous_service_name)
+        label_operations = [adwords_operation for adwords_operation in adwords_operations if
+                            'labelId' in adwords_operation['operand']]
+        regular_operations = [adwords_operation for adwords_operation in adwords_operations if
+                              'labelId' not in adwords_operation['operand']]
+        if len(label_operations) > 0:
+            results.append(service.cs_mutate_labels(previous_client_id, label_operations))
+        if len(regular_operations) > 0:
+            results.append(service.cs_mutate(previous_client_id, regular_operations))
+
+        self._operations_buffer = None
+        # maybe clean buffer?
+        # raise NotImplementedError()
+        return results
+        # must get, maybe process, and return results
+        # raise NotImplementedError()
 
     # TODO: this method should instantiate a new class (maybe SyncOperation) and transform the internal functions
     # into instance methods. Also, separate the treatment for each "object_type" into a new method as well.
-    def execute_operations(self, operations_folder):
+    def execute_operations(self, operations_folder=None, sync=False):
         """
         Possible columns in the table:
 
         :param table_name:
         :param batchlog_table:
         :return:
         """
-        logger.info('Running %s...', inspect.stack()[0][3])
-        _, files = self.storage.listdir(operations_folder)
-        files = [[path.join(operations_folder, file)] for file in files]
-        self._client = None
-        self._operations_buffer = None
-        self.map_function(self._batch_operations, files)
+        if sync:
+            return self._sync_operations()
+            # call self._sync_operations straight away or use map_function?
+        else:
+            if not operations_folder:
+                raise ValueError('Async operations must have an operation folder defined.')
+            logger.info('Running %s...', inspect.stack()[0][3])
+            _, files = self.storage.listdir(operations_folder)
+            files = [[path.join(operations_folder, file)] for file in files]
+            self._client = None
+            self._operations_buffer = None
+            self.map_function(self._batch_operations, files)
 
     def get_accounts(self, client_id=None):
-        mcs = ManagedCustomerService(self.client)
-        return {k[1]: v[1] for k, v in mcs.get_customers(client_id)}
+        mcs = self.service('ManagedCustomerService')
+        return {account['name']: account for account in mcs.get_customers(client_id)}
+
+    def get_entities(self, get_internal_operation):
+        try:
+            service_name = self._get_service_from_object_type(get_internal_operation)
+            service = self.service(service_name)
+            results = service.cs_get(get_internal_operation)
+            return [result for result in results]
```

Comment:
Hein? `return [result for result in results]`

`return results` ou até `return list(results)`